From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MC20018 <winme2001@outlook.com>
Date: Thu, 4 Jun 2026 17:54:00 +0800
Subject: [PATCH] Fand: hello world log line at server boot


diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index ff805b2496baea7b1644d1b5d9e0f2939aea245d..be7ddd48c71592e21bd793e04a0668ab891a7c78 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -64,6 +64,7 @@ public class Main {
 
     @SuppressForbidden(reason = "System.out needed before bootstrap")
     public static void main(final String[] args) {
+        LogUtils.getLogger().info("[Fand] Hello from a Fand patch! Booting on top of vanilla.");
         SharedConstants.tryDetectVersion();
         OptionParser parser = new OptionParser();
         OptionSpec<Void> nogui = parser.accepts("nogui");
