From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MC20018 <winme2001@outlook.com>
Date: Thu, 4 Jun 2026 17:56:17 +0800
Subject: [PATCH] Attach Fand runtime to vanilla server loop


diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index 45af5e7b74264804eb596052823a89c4acc62f2c..a220de69de1614848df32451ef7831c37e8e8aae 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -227,6 +227,7 @@ public class Main {
                         server.showGui();
                     }
 
+                    io.fand.server.Main.runtime().attach(server);
                     return server;
                 }
             );
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 6d5568909edfa94adaf61e684378cbb81e5326e5..d162ee94c66fd320906272764d700d5b46e4d188 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1015,6 +1015,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         profiler.push("scheduledPacketProcessing");
         this.packetProcessor.processQueuedPackets();
         profiler.pop();
+        io.fand.server.Main.runtime().tickScheduler();
         this.tickServer(sprinting ? () -> false : this::haveTime);
         this.tickFrame.end();
         profiler.pop();
