From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Fand <fand@local>
Date: Sat, 6 Jun 2026 15:54:58 +0800
Subject: [PATCH] Use TerminalConsoleAppender for server console


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index d5044deadbd8c390d0271bc27a3fbb12710dabd3..7e22ba953f78c74793b9b7f2dc5adb91e4ddc7c1 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -167,6 +167,9 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         Thread consoleThread = new Thread("Server console handler") {
             @Override
             public void run() {
+                // Fand start - Use TerminalConsoleAppender
+                new io.fand.server.console.FandConsole(DedicatedServer.this).start();
+                /*
                 BufferedReader reader = new BufferedReader(new InputStreamReader(System.in, StandardCharsets.UTF_8));
 
                 String line;
@@ -177,6 +180,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
                 } catch (IOException e) {
                     DedicatedServer.LOGGER.error("Exception handling console input", e);
                 }
+                */
+                // Fand end - Use TerminalConsoleAppender
             }
         };
         consoleThread.setDaemon(true);
