From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MC20018 <winme2001@outlook.com>
Date: Tue, 9 Jun 2026 22:45:14 +0800
Subject: [PATCH] Optimize explosion block position traversal


diff --git a/net/minecraft/world/level/ServerExplosion.java b/net/minecraft/world/level/ServerExplosion.java
index 8fead1dc8bfe84cd124d666b861ccdf6ffd7bd6a..7f9dbf69b5f75aea7ff2d573e5a13259853e0b56 100644
--- a/net/minecraft/world/level/ServerExplosion.java
+++ b/net/minecraft/world/level/ServerExplosion.java
@@ -120,7 +120,7 @@ public class ServerExplosion implements Explosion {
 
     private List<BlockPos> calculateExplodedPositions() {
         Set<BlockPos> toBlowSet = new HashSet<>();
-        int size = 16;
+        BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos();
 
         for (int xx = 0; xx < 16; xx++) {
             for (int yy = 0; yy < 16; yy++) {
@@ -137,10 +137,9 @@ public class ServerExplosion implements Explosion {
                         double xp = this.center.x;
                         double yp = this.center.y;
                         double zp = this.center.z;
-                        float stepSize = 0.3F;
 
                         while (remainingPower > 0.0F) {
-                            BlockPos pos = BlockPos.containing(xp, yp, zp);
+                            pos.set(xp, yp, zp);
                             BlockState block = this.level.getBlockState(pos);
                             FluidState fluid = this.level.getFluidState(pos);
                             if (!this.level.isInWorldBounds(pos)) {
@@ -153,7 +152,7 @@ public class ServerExplosion implements Explosion {
                             }
 
                             if (remainingPower > 0.0F && this.damageCalculator.shouldBlockExplode(this, this.level, pos, block, remainingPower)) {
-                                toBlowSet.add(pos);
+                                toBlowSet.add(pos.immutable());
                             }
 
                             xp += xd * 0.3F;
